packages/google_sign_in/android/src/main/java/io/flutter/plugins/googlesignin/GoogleSignInPlugin.java

```diff
@@ -79,37 +84,33 @@
     }
   }
 
-  private final FragmentActivity activity;
+  private final Activity activity;
   private final BackgroundTaskRunner backgroundTaskRunner;
   private final int requestCode;
 
+  private boolean resolvingError = false; // Whether we are currently resolving a sign-in error
   private GoogleApiClient googleApiClient;
   private List<String> requestedScopes;
   private PendingOperation pendingOperation;
 
   public static void registerWith(PluginRegistry.Registrar registrar) {
     Activity activity = registrar.activity();
-    if (!(activity instanceof FragmentActivity)) {
-      throw new IllegalArgumentException(
-          GoogleSignInPlugin.class.getSimpleName()
-              + " requires your activity to be an instance of "
-              + FragmentActivity.class.getName()
-              + ". You may want to use "
-              + FlutterFragmentActivity.class.getName());
-    }
-    FragmentActivity fragmentActivity = (FragmentActivity) activity;
     final MethodChannel channel = new MethodChannel(registrar.messenger(), CHANNEL_NAME);
     final GoogleSignInPlugin instance =
-        new GoogleSignInPlugin(fragmentActivity, new BackgroundTaskRunner(1), REQUEST_CODE);
+        new GoogleSignInPlugin(activity, new BackgroundTaskRunner(1), REQUEST_CODE);
     registrar.addActivityResultListener(instance);
+    registrar.publish(instance);
```

Comment:
Do we plan on anything extracting this instance via `PluginRegistry.valuePublishedByPlugin()`?